packages/contracts-bedrock/src/governance/ProposalValidator.sol

```diff
@@ -755,24 +813,27 @@ contract ProposalValidator is OwnableUpgradeable, ReinitializableBase, ISemver {
             _buildApprovalModuleOptions(_optionsDescriptions, _optionsRecipients, _optionsAmounts);
 
         // Configure approval module settings
-        IApprovalVotingModule.ProposalSettings memory settings = IApprovalVotingModule.ProposalSettings({
-            maxApprovals: uint8(optionsLength),
+        IApprovalVotingModule.ProposalSettings memory approvalSettings = IApprovalVotingModule.ProposalSettings({
+            maxApprovals: uint8(_optionsDescriptions.length),
             criteria: uint8(IApprovalVotingModule.PassingCriteria.Threshold),
             budgetToken: Predeploys.GOVERNANCE_TOKEN,
             criteriaValue: _criteriaValue,
             budgetAmount: uint128(totalBudget)
```

Comment:
I still believe that we should be safe about downcasting without checking the values. This `totalBudget` can actually underflow. As an example, see this in chisel: `uint128(uint256(0xf000000000000000000000000000000000000000000000000000000000000001))`

Adding checks doesn't hurt at all.